sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidJoinRule.java

```diff
@@ -153,7 +154,7 @@ public void onMatch(RelOptRuleCall call)
       leftFilter = null;
     }
 
-    if (!plannerContext.getJoinAlgorithm().requiresSubquery()
+    if (!QueryUtils.getJoinAlgorithm(join, plannerContext).requiresSubquery()
```

Comment:
note: I feel like both `PlannerContext` and `QueryUtils` is a bit unfortunate place for these....
have you considered placing it in `JoinAlgorithm`? 

you should probably also extract a `JoinAlgorithm` to a local variable

Reply:
Will move this out from QueryUtils in next batch of changes where I need to update the method as well.